packages/plugins/i18n/server/services/entity-service-decorator.js

```diff
@@ -164,13 +165,14 @@ const decorator = (service) => ({
       return service.findMany.call(this, uid, opts);
     }
 
-    const { kind } = strapi.getModel(uid);
+    const { kind } = model;
 
     if (kind === 'singleType') {
       if (opts[LOCALE_QUERY_FILTER] === 'all') {
         // TODO Fix so this won't break lower lying find many wrappers
-        const wrappedParams = await this.wrapParams(opts, { uid, action: 'findMany' });
-        return strapi.db.query(uid).findMany(wrappedParams);
+        const wrappedParams = await this.wrapParams(opts, { uid, action: 'findMany' }) || {};
+        const query = transformParamsToQuery(model, wrappedParams);
```

Comment:
transformParamsToQuery should receive a uid string no?

Reply:
When I give it a uuid string it complains about not being able to find attributes

Reply:
does it work though? can you populate fields?

Reply:
Will test it

Reply:
Tested it and no more error.

Reply:
It doesn't error out and in some cases by luck it works, but transfomrParamsToQuery should receive a string with the model uid.

```js
const transformParamsToQuery = (uid, params) => {
  const schema = strapi.getModel(uid);
```

The resulting schema is undefined if you send the model object,  and in some cases it works, but there are some params (filters, some nested populate, ...),that tneed the schema to be computed.

Reply:
This filter doesn't work with this current implementation

![image](https://github.com/strapi/strapi/assets/20578351/36ab5785-0e71-40fa-b14d-7606e4649f9f)